### PR TITLE
Add Windows and Codecov to Travis CI

### PR DESCRIPTION
**Pull request by mmiller-max**
Testing code coverage

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -156,5 +157,13 @@ end
     @test ts_reporting_copy.results == ts_reporting.results
 end
 
+@testset "showerror" begin
+    @test_throws TestReports.PkgTestError throw(TestReports.PkgTestError("Test"))
+    io = IOBuffer()
+    showerror(io, TestReports.PkgTestError("Error text"), "")
+    @test String(take!(io)) == "Error text"
```
Comment (oxinabox) [inline test/runtests.jl:164]: i think this can be done with `sprint` so don't need the `IOBuffer`

Comment (mmiller-max) [reply test/runtests.jl]: Thanks, good suggestion

**Comment by mmiller-max** [comment]
All there from a code coverage point of view, only two lines reported as not being covered and this is because of the v1.0.5 tests, when only the [v1.4 tests finished](https://codecov.io/gh/JuliaTesting/TestReports.jl/tree/63dd658558d53ce5aeb0c8c273c25362bf98fca0/src) they were compiled away.

Just need to work out why the Windows 1.4 build is failing. Once working I'll remove the AppVeyor yaml.

**Comment by mmiller-max** [comment]
OK this becomes a bit bigger than I expected, as a few bugs were found along the way. Summary of changes:

- Add code coverage to yaml
- Move Windows CI to Travis (and delete `appveyor.yml`)
- Refactor `gettestfilepath` so that coverage ignores code for other Julia versions, and also makes `test!` neater
- Fix bugs in `gettestfilepath` for v1.0.5 and v1.2
- Fix file path bug for Windows
- Add new tests to improve coverage
- Update README with correct tags

@oxinabox if you could please have a final look, I'll leave off merging until you've confirmed that you're happy with these changes as a fair bit has changed since your earlier review. Cheers

**Comment by mmiller-max** [comment]
Not sure why the build is not reporting correctly here, but it has completed fine.

**Comment by mmiller-max** [comment]
@oxinabox are you happy with these changes? It'd good to merge this!

**Pull request merged**